plugins/automation/lib/discourse_automation/event_handlers.rb

```diff
@@ -44,6 +49,11 @@ def self.handle_post_created_edited(post, action)
             next if !category_ids.include?(restricted_category["value"])
           end
 
+          restricted_tags = automation.trigger_field("restricted_tags")
+          if restricted_tags["value"]
+            next if (restricted_tags["value"] & topic.tags.map(&:name)).empty?
```

Comment:
Maybe we could use `intersect?` [here](https://stackoverflow.com/a/67493003/14880382)?

Or `any?` that I think is also [a good option](https://stackoverflow.com/a/33615934/14880382)